packages/create-start/src/modules/vscode/index.ts

```diff
@@ -0,0 +1,37 @@
+import { dirname } from 'node:path'
+import { fileURLToPath } from 'node:url'
+import { z } from 'zod'
+import { initHelpers } from '../../utils/helpers'
+import { createModule } from '../../module'
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = dirname(__filename)
+
+export const vsCodeModule = createModule(z.object({}))
+  .init((schema) => schema)
+  .prompt((schema) => schema)
+  .validateAndApply({
+    validate: async ({ targetPath }) => {
+      const _ = initHelpers(__dirname, targetPath)
+
+      return await _.getTemplateFilesThatWouldBeOverwritten({
+        file: '**/*',
+        templateFolder: './template',
+        targetFolder: targetPath,
+        overwrite: false,
+      })
+    },
+    apply: async ({ targetPath }) => {
+      // Copy the vscode template folders into the project
+      const _ = initHelpers(__dirname, targetPath)
+
+      // TODO: Handle when the settings file already exists and merge settings
```

Comment:
when would that file exist already?

Reply:
In the case the module is called when adding something to an existing project where that module needs to add vscode settings. This might not be called just for router vscode settings for example